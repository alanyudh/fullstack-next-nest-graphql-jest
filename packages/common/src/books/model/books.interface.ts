export interface IBook {
  id: string;
  title: string;
  description: string;
  year: number;
}
