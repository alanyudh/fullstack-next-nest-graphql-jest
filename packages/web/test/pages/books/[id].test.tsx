import { render } from '../../testUtils'
import Page from '../../../pages/books/[id]'
import { Book } from '../../../modules/books/models/book.model'

describe('Book detail page', () => {
  it('matches snapshot', () => {
    const props = {
      book: new Book('1', 'title1', 'description1', 2001),
    }

    const { asFragment } = render(<Page />, props)
    expect(asFragment()).toMatchSnapshot()
  })
})
