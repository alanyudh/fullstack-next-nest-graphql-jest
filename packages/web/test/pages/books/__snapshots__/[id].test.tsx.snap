// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book detail page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-row"
    style="margin-top: 32px; text-align: center;"
  >
    <div
      class="ant-col ant-col-24"
    >
      <article
        class="ant-typography"
      >
        <h1
          class="ant-typography"
          data-testid="book-detail-title"
        />
        <div
          class="ant-typography"
          data-testid="book-detail-desc"
        />
        <div
          class="ant-typography"
        >
          <span
            class="ant-typography ant-typography-secondary"
            data-testid="book-detail-year"
          >
            Published on 
          </span>
        </div>
        <div
          class="ant-divider ant-divider-horizontal"
          role="separator"
        />
        <a
          class="ant-typography"
          href="/"
        >
          back
        </a>
      </article>
    </div>
  </div>
</DocumentFragment>
`;
